src/utils/item.test.ts

```diff
@@ -0,0 +1,181 @@
+import { DiscriminatedItem, PermissionLevel } from '@graasp/sdk';
+
+import { describe, expect, it } from 'vitest';
+
+import {
+  getHighestPermissionForMemberFromMemberships,
+  getParentsIdsFromPath,
+  isUrlValid,
+  transformIdForPath,
+} from './item';
+
+describe('item utils', () => {
+  it('isUrlValid', () => {
+    expect(isUrlValid(null)).toBeFalsy();
+    expect(isUrlValid()).toBeFalsy();
+    expect(isUrlValid('somelink')).toBeFalsy();
+    expect(isUrlValid('graasp.eu')).toBeTruthy();
+    expect(isUrlValid('https://graasp')).toBeFalsy();
+
+    expect(isUrlValid('https://graasp.eu')).toBeTruthy();
+    expect(isUrlValid('http://graasp.eu')).toBeTruthy();
+    expect(isUrlValid('https://www.youtube.com/')).toBeTruthy();
+  });
+
+  it('transformIdForPath', () => {
+    expect(transformIdForPath('someid')).toEqual('someid');
+    expect(transformIdForPath('some-id')).toEqual('some_id');
+    const id = 'ecafbd2a-5688-11eb-ae93-0242ac130002';
+    const path = 'ecafbd2a_5688_11eb_ae93_0242ac130002';
+    expect(transformIdForPath(id)).toEqual(path);
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
+    expect(() => transformIdForPath(null)).toThrow();
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
+    expect(() => transformIdForPath(undefined)).toThrow();
+  });
+
+  describe('getParentsIdsFromPath', () => {
```

Comment:
I think this exists in sdk.

Reply:
Yes I will remove them in another PR, as during my first try a lot of tests broke, and I did not want to mix the bug fix and the refactoring.